pkg/syncer/metadatasyncer.go

```diff
@@ -330,6 +342,26 @@ func InitMetadataSyncer(ctx context.Context, clusterFlavor cnstypes.CnsClusterFl
 				return logger.LogNewErrorf(log, "failed to start informer on %q instances. Error: %v",
 					cnsoperatorv1alpha1.CnsStoragePolicyQuotaSingular, err)
 			}
+
+			go func() {
+				isStorageQuotaM2Enabled := metadataSyncer.coCommonInterface.IsFSSEnabled(ctx, common.StorageQuotaM2)
+				if isStorageQuotaM2Enabled {
+					// TODO: Wait for fullSync to update "zones" value on all CnsVolumeInfo CRs
+
+					// Start informer on AvailabilityZone CRs
+					err = startAvailabilityZoneCRInformer(ctx, k8sConfig)
+					if err != nil {
+						if err == common.ErrAvailabilityZoneCRNotRegistered {
```

Comment:
This is an exit-worthy error; let's error out here, not just log.

Reply:
In latest supervisor clusters, we will not face this issue. This is unlikely case, but for safer side added this check to silently log the info message.

Reply:
We should not face this issue on new supervisor, so erroring out for this case.